src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -0,0 +1,250 @@
+        throw new UnsupportedOperationSQLException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, String[] columnNames) throws SQLException {
+        throw new UnsupportedOperationSQLException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // SQL99 data types
+
+    @Override
+    public Clob createClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Blob createBlob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public NClob createNClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public SQLXML createSQLXML() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Struct createStruct(String typeName, Object[] attributes) throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Database meta data
+
+    @Override
+    public DatabaseMetaData getMetaData() throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-37");
+    }
+
+    /**
+     * Used during Hibernate's DDL step for Information Extraction purposes.
+     *
+     * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
+     */
+    @Override
+    public @Nullable String getCatalog() {
+        return null;
+    }
+
+    /**
+     * Used during Hibernate's DDL step for Information Extraction purposes.
+     *
+     * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
+     */
+    @Override
+    public @Nullable String getSchema() {
+        return null;
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // unsupported operations
+
+    // stored procedure equivalence in MongoDB won't be supported
+    // see https://www.mongodb.com/resources/products/capabilities/stored-procedures
+
+    @Override
+    public CallableStatement prepareCall(String mql) throws SQLException {
+        throw new UnsupportedOperationSQLException("Stored procedure equivalence in MongoDB unsupported");
+    }
+
+    @Override
+    public CallableStatement prepareCall(String mql, int resultSetType, int resultSetConcurrency) throws SQLException {
+        throw new UnsupportedOperationSQLException("Stored procedure equivalence in MongoDB unsupported");
+    }
```

Comment:
It is a huge reliever that our product doesn't need to be concerned about `stored procedure` emulation.